### DISCUSSION TITLE
Table data search but triggered by button instead

### DISCUSSION THREAD

**Discussion by 0afcode** [Budibase/budibase#10402] [Help] [answered]
Hi, I am following the tutorial on this page ( https://docs.budibase.com/docs/adding-a-search-field ) but essentially what I want is to do option 2, but instead of auto refresh when someone types on the fields, trigger the search only when a form button is clicked. How can I do this?

**Comment by melohagan** [comment, marked as answer]
Hey @0afcode 

If you have followed the steps in option 2, then this is fairly easy to do!

### Step 1 - Add your search button

Add a container and nest your search field within it. Next add a button. You can set the **direction** of the [container](https://docs.budibase.com/docs/container) to get the button to sit nicely beside the text field.

![Screenshot 2023-04-24 at 17 33 53](https://user-images.githubusercontent.com/101575380/234059903-715d0a7a-66a0-4480-a757-f7394566d877.png)

Next `Define actions` for your button. Add the **Update State** action, with a key and value bound to the search field:

![Screenshot 2023-04-24 at 17 35 11](https://user-images.githubusercontent.com/101575380/234060153-1c996f9b-780b-416c-86f4-addf6c17ac47.png)

### Step 2 - Filtering on app state

We can now use the [App State](https://docs.budibase.com/docs/app-state) that we have saved to filter on. Because the app state is only updated on the button click in this case, the filter will only trigger when the button is clicked.

 Click `Define filters` on the Data provider of the table, and use the following filter:

![Screenshot 2023-04-24 at 17 38 34](https://user-images.githubusercontent.com/101575380/234060889-0dea1f91-3c72-47f5-bf02-831a5008ee06.png)

That's it!

### Result
![search](https://user-images.githubusercontent.com/101575380/234061148-3bd3b84e-14ce-4648-b3db-e44b3855d8b0.gif)

**Reply by 0afcode** [reply]
perfect, thank you
